mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -1593,3 +1593,40 @@ void linalg::populateDecomposeConvolutionPatterns(RewritePatternSet &patterns,
       DownscaleSizeOneWindowed2DConvolution<PoolingNchwMaxOp, PoolingNcwMaxOp>>(
       patterns.getContext(), benefit);
 }
+
+Value mlir::linalg::createReadOrMaskedRead(OpBuilder &builder, Location loc,
+                                           Value source,
+                                           ArrayRef<int64_t> readShape,
+                                           Value padValue, bool doMasking) {
+  assert(llvm::none_of(readShape,
+                       [](int64_t s) { return s == ShapedType::kDynamic; }));
```

Comment:
nit: add `&& "expected static shape"` for better error messages

Reply:
Fixed.